### PR TITLE
Implement true random number generator on posix platform.

### PR DESCRIPTION
**Pull request by jwhui**

### CODE REVIEW

File: examples/platforms/posix/random.c [hunk 1]
```diff
@@ -75,6 +88,40 @@ ThreadError otPlatRandomSecureGet(uint16_t aInputLength, uint8_t *aOutput, uint1
...
+
+    *aOutputLength = 0;
+
+    otEXPECT_ACTION(aOutput && aOutputLength, error = kThreadError_InvalidArgs);
+
+    file = fopen("/dev/urandom", "rb");
+    otEXPECT_ACTION(file != NULL, error = kThreadError_Failed);
+
+    readLength = fread(aOutput, 1, aInputLength, file);
+    otEXPECT_ACTION(readLength == aInputLength, error = kThreadError_Failed);
```
Comment (bukepo) [inline examples/platforms/posix/random.c]: In the declaration, aOutputLength may be less than aInputLength. Shall we just fail here?

Comment (jwhui) [inline examples/platforms/posix/random.c]: Good point.  The declaration was modeled off of `mbedtls_hardware_poll()`.  However, mbedtls also errors when the desired length cannot be generated.  I'm thinking of updating the documentation (and possibly this API) to simply return error if the requested length cannot be provided.  Thoughts?

Comment (bukepo) [inline examples/platforms/posix/random.c]: Agree, I think that would be cleaner.

Comment (jwhui) [inline examples/platforms/posix/random.c]: Sounds good, I'll work on that in a separate PR.

**Pull request merged**